configuration/nfs.md

```diff
@@ -0,0 +1,303 @@
+/export/users 192.168.1.0/24(rw,nohide,insecure,no_subtree_check,async)
+```
+
+### Portmap lockdown (optional)
+The files on your NFS are open to anyone on the network. As a security measure, you can restrict access to specified clients.
+
+Add the following line to `/etc/hosts.deny`:
+
+```
+rpcbind mountd nfsd statd lockd rquotad : ALL
+```
+
+By blocking all clients first, only clients in `/etc/hosts.allow` (added below) will be allowed to access the server. 
+
+Now add the following line to `/etc/hosts.allow`:
+
+```
+rpcbind mountd nfsd statd lockd rquotad : <list of IPv4s>
+```
+
+Where `<list of IPv4s>` is a list of IP addresses that consists of the server and all clients. (These have to be IP addresses because of a limitation in `rpcbind`, which it doesn't like hostnames.) Note that if you have NIS set up, you can just add these to the same line.
+
+Please ensure that the list of authorised IP addresses includes the `localhost` address (`127.0.0.1`), as the startup scripts in recent versions of Ubuntu use the `rpcinfo` command to discover NFSv3 support, and this will be disabled if `localhost` is unable to connect.
+
+Finally, to make your changes take effect, restart the service:
+
+```bash
+sudo systemctl restart nfs-kernel-server
+```
+
+## Setup a NFSv4 client
+
+Now that your server is running, you need to set up any clients to be able to access it. To start, install the required packages:
+
+```bash
+sudo apt install nfs-common
+```
+
+On the client, we can mount the complete export tree with one command:
+
+```bash
+mount -t nfs -o proto=tcp,port=2049 <nfs-server-IP>:/ /mnt
+```
+
+You can also specify the NFS server hostname instead of its IP address, but in this case you need to ensure that the hostname can be resolved to an IP on the client side. A robust way of ensuring that this will always resolve is to use the `/etc/hosts` file.
+
+Note that `<nfs-server-IP>:/export` is not necessary in NFSv4, as it was in NFSv3. The root export `:/` defaults to export with `fsid=0`.
+
+We can also mount an exported subtree with:
+
+```bash
+mount -t nfs -o proto=tcp,port=2049 <nfs-server-IP>:/users /home/users
+```
+
+To ensure this is mounted on every reboot, add the following line to `/etc/fstab`:
+
+```
+<nfs-server-IP>:/   /mnt   nfs    auto  0  0
+```
+
+If after mounting, the entry in `/proc/mounts appears` as `<nfs-server-IP>://` (with two slashes), then you might need to specify two slashes in `/etc/fstab`, or else `umount` might complain that it cannot find the mount.
+
+### Portmap lockdown (optional)
+
+Add the following line to `/etc/hosts.deny`:
+
+```
+rpcbind : ALL
+```
+
+By blocking all clients first, only clients in `/etc/hosts.allow` below will be allowed to access the server.
+
+Now add the following line to `/etc/hosts.allow`:
+
+```
+rpcbind : <NFS server IP address>
+```
+
+Where `<NFS server IP address>` is the IP address of the server.
+
+## NFS Server with complex user permissions
```

Comment:
Camel caps is wonky. Make the 's' on server lower case to match rest of sentence